packages/style-engine/class-wp-style-engine.php

```diff
@@ -423,20 +445,25 @@ public function get_block_supports_styles( $block_styles, $options ) {
 		}
 
 		// Build CSS rules output.
-		$css_selector = isset( $options['selector'] ) ? $options['selector'] : null;
-		$style_rules  = new WP_Style_Engine_CSS_Declarations( $css_declarations );
+		$css_selector        = isset( $options['selector'] ) ? $options['selector'] : null;
+		$should_enqueue      = isset( $options['enqueue'] ) && true === $options['enqueue'];
+		$wp_css_declarations = new WP_Style_Engine_CSS_Declarations( $css_declarations );
 
 		// The return object.
 		$styles_output = array();
-		$css           = $style_rules->get_declarations_string();
+		$css           = $wp_css_declarations->get_declarations_string();
 
 		// Return css, if any.
 		if ( ! empty( $css ) ) {
 			$styles_output['css']          = $css;
-			$styles_output['declarations'] = $style_rules->get_declarations();
+			$styles_output['declarations'] = $wp_css_declarations->get_declarations();
 			// Return an entire rule if there is a selector.
 			if ( $css_selector ) {
 				$styles_output['css'] = $css_selector . ' { ' . $css . ' }';
+				if ( $should_enqueue ) {
+					$current_css_rule = static::$block_supports_store->get_rule( $css_selector );
```

Comment:
Here is where it'd be nice to have an explicit `set_rule` method.

In come cases we want to return a css rule string, e.g., 

```php
$styles_output['css'] = $css_selector . ' { ' . $css . ' }';
```

and not store it.

With a `set_rule` method we could do the following:

```php
if ( $css_selector ) {
    $css_rule    = new WP_Style_Engine_CSS_Rule( $css_selector, $css_declarations )
    $styles_output['css'] = $css_rule-get_css();
    if ( $should_enqueue ) {
          $current_css_rule = static::$block_supports_store->get_rule( $css_selector );
          $current_css_rule->set_declarations( $css_rule  );
      }
}
```